FAQ additions (#101)
stackblitz/docs: docs/guides/user-guide/general-faqs.md

## Enterprise Server

### What is the Enterprise Server offering?

Enterprise Server is a membership offering that's best fit for organizations who want to use StackBlitz behind their firewall. Benefits include:

- Private NPM registry integration
- Custom SSO integration
- On-prem installation support
- Dedicated Customer Success Engineer

Check out the [Enterprise page](https://stackblitz.com/enterprise) for details and set up a call with our Enterprise team!

### I want to purchase the enterprise edition for WebContainer. Where can I do that?

Bringing Webcontainer to Enterprise Server is in the works, but not currently available. You can reach out to enterprise@stackblitz.com for details.

We do offer [WebContainer API](webcontainers.io) licenses for organizations wanting to use WebContainer technology for production usage in a for-profit setting. Check out [webcontainers.io](webcontainers.io) for more information!

# Technology FAQs

## Classic Editor (Engineblock projects)

### What is the Classic Editor for Engineblock projects?

> The Classic Editor utilizes Engineblock and WeBcontainer technologies to run. As the original StackBlitz interface, the Classic Editor has a file tree, an npm package manager, and a light dev server and preview pane that allows you to see your app as you code.

Author: Rewrite sentence - interpretted as both technologies are needed to run.

> Engineblock projects are distinguishable as they do not have a terminal.

Author: distinguishable FROM webcontainers-based projects. for more info re: webcontainers, <ADD LINK that goes down to webcontainer FAQ block>

### How do I create code snippets?

Code snippets are available in our Classic Editor for both Engineblock and WebContainer projects. Click on the Settings icon in the activity bar to the left of your screen. The Settings sidebar will pop open and you will see User snippets and Workspace snippets. User snippets are available for any of your projects. Workspace snippets are only available in the current project you create them in.

### Can I change the TypeScript version in my project?

The current TypeScript starter uses EngineBlock technology, which has a fixed TypeScript version. EngineBlock is great for lightweight frontends, but does not run Node.js itself and will be limited to things like TypeScript versioning.

Webcontainer, however does run Node.js itself in the browser and does not have this limitation. You can simply `npm install` any TypeScript version as you would locally.

## Classic Editor (WebContainer projects)

### What is the Classic Editor for WebContainer projects?

> The Classic Editor utilizes Engineblock and WeBcontainer technologies to run. As the original StackBlitz interface, the Classic Editor has a file tree, an npm package manager, and a light dev server and preview pane that allows you to see your app as you code.

Author: rewrite sentence - reads as if BOTH technologies are needed to run classic editor, but not true.

Any projects in `https://stackblitz.com/?starters=fullstack`, `..?starters=backend`, `..?starters=vite` will be WebContainer/Node projects with a terminal.

### Can I change the Node version in my project?

The current state of Codeflow does not support users being able to set their own Node version.

### Why can't I see my terminal?

Please note that the terminal is only available for WebContainer-based projects. You can use the keyboard binding `^` `` ` `` (the caret and the back quote keys) to toggle the terminal.

### Can I run unit tests in StackBlitz?

You can use [Jasmine](https://jasmine.github.io/), a unit testing tool for JavaScript.

It is also possible to run Angular unit tests inside WebContainer-based projects, but not Engineblock-based projects. The easiest way to start is by navigating to node.new, which creates a WebContainer-based project automatically!

### Can I develop ElectronJS apps on StackBlitz?

Since Electron is a native app, this cannot be supported since StackBlitz runs entirely within the browser.

## Codeflow Editor

### What is the Codeflow Editor?

> The Codeflow Editor utilizes our Codeflow technology to bring VSC code into the browser. This editor is capable of running full development workflows, including pushing commits and creating pull requests to GitHub. This bypasses the need to clone a repository or install dependencies on your local machine. If you haven't already, you can check out Codeflow [here](https://stackblitz.com/codeflow)!

Author: rewrite as: "The Codeflow Editor brings VS Code into the browser."

Author: after "...creating pull requests to GitHub", add the features as a separate sentence.

### Can I use Vim bindings?

Our Codeflow editor is built on top of VS Code so you can enable the VIM extension:

1. Open a github repository in Codeflow (like this: pr.new/vitejs/vite)
2. Navigate to the "Extensions" pane
3. Search for "Vim"
4. Click on the VIM extension and then click the "enable" button.

## Web Publisher

### What is Web Publisher?

Web Publisher is a single file editor and publishing tools that can be used for smaller or text-based changes such as editiong documentation or blogs. Web Publisher uses Codeflow technology that allows users to create, push, and publish changes all in their browser. The GitHub integration creates a seamless workflow that's fit for developers and non-coding folks alike!

This environment is ideal for:
- No-tech or low-tech users
- Text or formatting changes in documentation or blogs
- Quick fixes that don't require dependency or compilation changes

You can learn more about Web Publisher [here](https://developer.stackblitz.com/codeflow/content-updates-with-web-publisher).

## Classic Editor SDK

### What is the Classic Editor SDK?

> Our SDK (software development kit) allows users to create StackBlitz projects to be opened in a new window or embedded in your docs, example pages, or blog posts. Projects are controlled by using the SDK's VM (virtual machine) interface. The VM can be used to control the UI of an embedded StackBlitz project, change the currently open file(s), and read and write files frmo the project's virtual filesystem.

Author: Our JavaScript-based SDK..... allows users to created SB Classic Editor projects.

Remove VM info. "The SDK can be used to control the UI......"

---
title: &title General FAQs (Frequently Asked Questions)
description: &description This page outlines commonly asked questions and answers about StackBlitz.
head:
  - ['meta', {property: 'og:title', content: *title}] 
  - ['meta', {property: 'og:image', content: 'https://developer.stackblitz.com/img/og/general-faqs.png'}]
  - ['meta', {name: 'twitter:title', content: *title}]
  - ['meta', {name: 'twitter:description', content: *description}]
---

This page outlines commonly asked questions and answers about StackBlitz.

# General FAQs

> Here's a [handy video](https://twitter.com/sulco/status/1275039159611203585) featured on Twitter that shows you how to create and use code snippets in StackBlitz.

Author: Yes! Thank you for catching this. Added the question into my updated commit. Thanks!

### Can I use Postman or something similar for testing?

[Postman](https://www.postman.com/automated-testing/) and similar tools will not work since the server is unreachable outside of the browser. As an alternative, you could run a node program that makes requests for testing.

### What are the deploy options available on StackBlitz? 

The three pathways we current support are:

1. If you are in any non-Codeflow StackBlitz project, you can download the project to your local machine by clicking the button right underneath the "Share" dialogue and deploy manually.

2. If you are using an EngineBlock project, you can deploy directly via our [Firebase integration](https://developer.stackblitz.com/guides/user-guide/ide-whats-on-your-screen#firebase-sidebar).

3. If you are using Codeflow, you can push code directly into GitHub and then use the [Netlify](https://github.com/apps/netlify) or [Vercel](https://vercel.com/docs/concepts/git/vercel-for-github) GitHub bots to deploy your site.

> 

Author: Thanks for pointing this out! Added this to my latest commit.

## Enterprise Server

### What is the Enterprise Server offering?

Enterprise Server is a membership offering that's best fit for organizations who want to use StackBlitz behind their firewall. Benefits include:

- Private NPM registry integration
- Custom SSO integration
- On-prem installation support
- Dedicated Customer Success Engineer

Check out the [Enterprise page](https://stackblitz.com/enterprise) for details and set up a call with our Enterprise team!

### I want to purchase the enterprise edition for WebContainer. Where can I do that?

Bringing Webcontainer to Enterprise Server is in the works, but not currently available. You can reach out to enterprise@stackblitz.com for details.

We do offer [WebContainer API](webcontainers.io) licenses for organizations wanting to use WebContainer technology for production usage in a for-profit setting. Check out [webcontainers.io](webcontainers.io) for more information!

# Technology FAQs

## Classic Editor (Engineblock projects)

### What is the Classic Editor for Engineblock projects?

> Different projects that use the Classic Editor utilize Engineblock technology to run. As the original StackBlitz interface, the Classic Editor has a file tree, an npm package manager, and a light dev server and preview pane that allows you to see your app as you code.

Reviewer: ```suggestion
Different projects that use the Classic Editor utilize [EngineBlock](/guides/user-guide/available-environments#engineblock) technology to run. As the original StackBlitz UI, the Classic Editor has a file tree, an npm package manager, and a lightweight dev server and preview pane that allows you to see your app as you code.
```

> Engineblock projects are distinguishable from WebContainer-based projects as they do not have a terminal.

Reviewer: ```suggestion
EngineBlock projects are distinguishable from WebContainers-based projects as they do not have a terminal.
```

### How do I create code snippets?

> Code snippets are available in our Classic Editor for both Engineblock and WebContainer projects. Click on the Settings icon in the activity bar to the left of your screen. The Settings sidebar will pop open and you will see User snippets and Workspace snippets. User snippets are available for any of your projects. Workspace snippets are only available in the current project you create them in.

Reviewer: ```suggestion
Code snippets are available in our Classic Editor for both EngineBlock and WebContainer projects. Click on the Settings icon in the activity bar to the [left of your screen](/guides/user-guide/ide-whats-on-your-screen#frontmatter-title). The Settings sidebar will pop open and you will see User snippets and Workspace snippets. User snippets are available for any of your projects. Workspace snippets are only available in the current project you create them in.
```

### Can I change the TypeScript version in my project?

The current TypeScript starter uses EngineBlock technology, which has a fixed TypeScript version. EngineBlock is great for lightweight frontends, but does not run Node.js itself and will be limited to things like TypeScript versioning.

> Webcontainer, however does run Node.js itself in the browser and does not have this limitation. You can simply `npm install` any TypeScript version as you would locally.

Reviewer: ```suggestion
WebContainers, however, do run Node.js itself in the browser and do not have this limitation. You can simply `npm install` any TypeScript version as you would locally.
```

> ## Classic Editor (WebContainer projects)

Reviewer: ```suggestion
## Classic Editor (WebContainers projects)
```

> ### What is the Classic Editor for WebContainer projects?

Reviewer: ```suggestion
### What is the Classic Editor for WebContainers projects?
```

> Different projects that use the Classic Editor utilize WebContainer technology to run. As the original StackBlitz interface, the Classic Editor has a file tree, an npm package manager, and a light dev server and preview pane that allows you to see your app as you code.

Reviewer: ```suggestion
Different projects that use the Classic Editor utilize WebContainers technology to run. As the original StackBlitz UI, the Classic Editor has a file tree, an npm package manager, and a light dev server and preview pane that allows you to see your app as you code.
```
